smartling-api-commons/src/main/java/com/smartling/api/v2/client/exception/RestApiRuntimeException.java

```diff
@@ -89,4 +92,17 @@ public String getMessage()
 
         return errorMessage.toString();
     }
+
+    private static Response fakeResponse(RequestContext requestContext)
+    {
+        if (requestContext != null && requestContext.getHeaders() != null)
+        {
+            return Response
+                .status(Response.Status.INTERNAL_SERVER_ERROR)
+                .header(REQUEST_ID_HEADER, requestContext.getHeaders().getFirst(REQUEST_ID_HEADER))
```

Comment:
I would put all headers into the response. There can be other headers that might be present in an exception handler or log helpfully.

Reply:
SDK originally takes requestId from response.
That is why I create a fake response and set requestId from request here - it covers cases when we got exception without response (e.g. SocketTimeoutException).